docs/content/DimensionSpecs.md

```diff
@@ -154,3 +154,195 @@ Example for the `__time` dimension:
   "function" : "function(t) { return 'Second ' + Math.floor((t % 60000) / 1000); }"
 }
 ```
+
+### Namespaced extraction function (EXPERIMENTAL)
+A namespaced extraction function is a simple key-value mapping where the key-value mappings are unique to a particular namespace.
+This can be used for re-naming values in a datasource for certain queries. To use a particular namespace, simply set the appropriate namespace for the dimExtractionFn
+```json
+    "dimExtractionFn" : {
+      "type":"namespace",
+      "namespace":"some_namespace"
+    }
+```
+Namespace updates can be set through adding a child in json form to the zookeeper path at`druid.zk.paths.namespacePath`. The json is of the following format:
+
+#### URI namespace update
+The remapping values for each namespace can be specified by a namespace update json object as per
+```json
+{
+  "namespace":{
+    "type":"uri",
+    "namespace":"some_namespace",
+    "uri": "file:///some/file/or/other/uri",
```

Comment:
Lets assume that a "get the latest URI" logic exists. If the FS is not atomic, then there is no guarantee that any particular node will see the latest URI when it needs to do the load. Which means it will either:

1. Detect the "latest" as the "some version before the latest" and load the wrong version if the node itself tries to detect the latest.
2. Try and force-fetch the "latest" and fail if the coordinator tries to force-tell the node what to load.

Hopefully I've missed a case here though. Or is the intent to do #2 and simply retry until it finds it?